cookbook/entities/acl.rst

```diff
@@ -0,0 +1,296 @@
+            *   class="AppBundle:Task",
+            *   permission="EDIT"
+            * )
+            */
+           public function editAction(Task $task, Request $request)
+           {
+               // ...
+           }
+       }
+
+   Using the ``@Acl`` annotation does not only create new access control lists to which you can
+   refer in other parts of your code it will also trigger the access decision manager when your
+   actions are accessed by users and thus protect them from being accessed without the needed
+   permissions.
+
+#. If you do not want to protect any controller methods or if you prefer to keep the definition of
+   your ACLs separated from the application code, you can define them using some YAML config in a
+   file named ``acl.yml``:
+
+   .. code-block:: yaml
+       :linenos:
+
+       # src/AppBunde/Resources/config/acl.yml
+       app_task_create:
+           type: entity
+           class: AppBunde\Entity\Task
+           permission: CREATE
+
+       app_task_delete:
+           type: entity
+           class: AppBunde\Entity\Task
+           permission: DELETE
+
+       app_task_edit:
+           type: entity
+           class: AppBunde\Entity\Task
+           permission: EDIT
+
+       app_task_view:
+           type: entity
+           class: AppBunde\Entity\Task
+           permission: VIEW
+
+.. sidebar:: Security Actions that Are not Related to an Entity
+
+    You can also create access control lists that are only used to protect certain actions that are
+    not related to an entity. To do that just set the type of the ACL to ``action`` instead of
+    ``entity``:
+
+    .. code-block:: php
+        :linenos:
+
+        // src/AppBunde/Controller/PageController.php
+        namespace AppBunde\Controller;
+
+        use Oro\Bundle\SecurityBundle\Annotation\Acl;
+        use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+        class PageController extends Controller
+        {
+            /**
+             * @Acl(
+             *     id="app_static_pages",
+             *     type="action"
+             * )
+             */
+            public function showAction()
+            {
+                // ...
+            }
+
+            // ...
+        }
+
+    When configuring the ACL using the YAML config format, you also have to set the controller to
+    bind the ACL to using the ``bindings`` option:
+
+    .. code-block:: yaml
+        :linenos:
+
+        # src/AppBunde/Resources/config/acl.yml
+        app_static_pages:
+            type: action
+            bindings:
+                class: AppBunde\Controller\PageController
+                method: showAction
+
+    .. seealso::
+
+        All configuration options are explained in full details in the :doc:`/reference/annotation/acl`,
+        :doc:`/reference/annotation/acl_ancestor`, and :doc:`ACL YAML format </reference/format/acl>`
+        reference.
+
+.. _coobook-entities-acl-check:
+
+Perform Access Checks
```

Comment:
Please add how to make protections with twig `resource_granted` function, with SecurityFacade in php code and ACL Helper for ORM queries